src/pip/_internal/utils/misc.py

```diff
@@ -510,6 +510,14 @@ def unzip_file(filename, location, flatten=True):
                 fn = split_leading_dir(name)[1]
             fn = os.path.join(location, fn)
             dir = os.path.dirname(fn)
+            # https://github.com/pypa/pip/issues/3907
+            if os.path.commonprefix(
+                    [os.path.abspath(location),
+                     os.path.abspath(dir)]) != os.path.abspath(location):
+                logger.warning(
```

Comment:
I'd suggest using `deprecated` helper (from utils.deprecation) for this, with a gone_in of `19.4` and no replacement.

Reply:
changes added